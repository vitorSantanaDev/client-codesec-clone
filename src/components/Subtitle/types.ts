import { ReactNode } from 'react'

export interface IViewProps {
  children: ReactNode
  color?: 'redGradient' | 'redDark' | 'redLight' | 'redPrimary'
  uppercase?: boolean
  icon?: boolean
  iconUrl?: string
}
