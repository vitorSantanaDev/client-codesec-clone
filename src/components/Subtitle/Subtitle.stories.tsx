import { ComponentStory, ComponentMeta } from '@storybook/react'
import Subtitle from '.'

export default {
  title: 'Subtitle',
  component: Subtitle,
  args: {
    children: 'Subtitle example',
    icon: true,
    iconUrl: '/images/icon-key.svg'
  }
} as ComponentMeta<typeof Subtitle>

export const Default: ComponentStory<typeof Subtitle> = (args) => (
  <Subtitle {...args} />
)
