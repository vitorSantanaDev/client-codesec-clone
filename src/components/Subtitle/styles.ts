import styled, { css } from 'styled-components'

type SubtitleProps = {
  color: 'redGradient' | 'redDark' | 'redLight' | 'redPrimary'
  uppercase: boolean
  icon: boolean
  iconUrl?: string
}

export const Wrapper = styled.span<SubtitleProps>`
  ${({ theme, color, uppercase, icon, iconUrl }) => css`
    background-image: ${theme.colors[color]};
    background-clip: text;
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;

    display: flex;
    align-items: center;

    font-weight: 700;
    letter-spacing: 15%;
    text-transform: ${uppercase ? 'uppercase' : 'none'};
    font-size: calc(${theme.font.sizes.small} + 0.2rem);

    ${icon
      ? `
      &::before {
        content: '';
        width: 24px;
        height: 24px;
        border-radius: 50%;
        display: inline-block;

        background-size: cover;
        background-image: url(${iconUrl});

        margin-right: calc(${theme.spacings.small} - 1rem);
      }
    `
      : ''}
  `}
`
