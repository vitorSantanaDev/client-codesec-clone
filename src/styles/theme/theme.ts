export const theme = {
  colors: {
    redPrimary: '#E83752',
    redLight: '#FA5770',
    redDark: '#D4203B',
    redError: '#EB3D3D',
    redGradient: 'linear-gradient(243.46deg, #FA5770 16.66%, #E83752 83.34%)',
    redGradientDark:
      'linear-gradient(225.05deg, #FA5770 0%, rgba(232, 55, 82, 0) 100%)',
    redGradientLight:
      'linear-gradient(225.05deg, rgba(250, 87, 112, 0.1) 0%, rgba(232, 55, 82, 0) 50%)',
    greenSucsses: '#32D957',
    white: '#FFFFFF',
    whiteLight: '#F5F0F1',
    whiteDark: '#EDE4E5',
    black: '#121111',
    blackLight: '#1C1B1B',
    blackDark: '#0A0909',
    gray1: '#141414',
    gray2: '#2C2C2C',
    gray3: '#444444',
    gray4: '#5C5C5C',
    gray5: '#747474',
    gray6: '#8C8C8C',
    gray7: '#A3A3A3'
  },
  font: {
    family: {
      default: "'Raleway', sans-serif"
    },
    sizes: {
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
      xxhuge: '8rem'
    }
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteLarge: '(max-width: 820px)',
    ltexLarge: '(max-width: 912px)',
    lteSmall: '(max-width: 420px)',
    lteXSmall: '(max-width: 280px)'
  },
  spacings: {
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
    xxhuge: '8rem'
  }
}
