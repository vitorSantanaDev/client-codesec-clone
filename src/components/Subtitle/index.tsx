import { ReactNode } from 'react'
import * as S from './styles'

type SubtitleProps = {
  children: ReactNode
  color?: 'redGradient' | 'redDark' | 'redLight' | 'redPrimary'
  uppercase?: boolean
  icon?: boolean
  iconUrl?: string
}

const Subtitle = ({
  children,
  color = 'redGradient',
  uppercase = true,
  icon = false,
  iconUrl
}: SubtitleProps) => (
  <S.Wrapper iconUrl={iconUrl} icon={icon} uppercase={uppercase} color={color}>
    {children}
  </S.Wrapper>
)

export default Subtitle
