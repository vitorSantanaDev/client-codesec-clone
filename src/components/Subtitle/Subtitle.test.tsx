import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import { theme } from 'styles/theme/theme'
import Subtitle from '.'

describe('<Subtitle />', () => {
  it('should render subtitle', () => {
    const { container } = renderTheme(<Subtitle>Subtitle example</Subtitle>)
    const subtitle = screen.getByText(/subtitle example/i)

    expect(subtitle).toBeInTheDocument()
    expect(subtitle).toHaveStyle({
      'font-weight': '700',
      'letter-spacing': '15%',
      'font-size': `calc(${theme.font.sizes.small} + 0.2rem)`
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
