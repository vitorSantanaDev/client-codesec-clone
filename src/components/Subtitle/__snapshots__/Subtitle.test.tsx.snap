// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Subtitle /> should render subtitle 1`] = `
.c0 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem + 0.2rem);
}

<span
  class="c0"
  color="redGradient"
>
  Subtitle example
</span>
`;
